packages/@aws-cdk/cloud-assembly-schema/schema/cloud-assembly.schema.json

```diff
@@ -0,0 +1,307 @@
+                        "$ref": "#/definitions/MissingContext"
+                    }
+                },
+                "runtime": {
+                    "description": "Runtime information. (Default - no info.)",
+                    "$ref": "#/definitions/RuntimeInfo"
+                }
+            },
+            "additionalProperties": false,
+            "required": [
+                "version"
+            ]
+        },
+        "ArtifactManifest": {
+            "description": "A manifest for a single artifact within the cloud assembly.",
+            "type": "object",
+            "properties": {
+                "type": {
+                    "$ref": "#/definitions/ArtifactType",
+                    "description": "The type of artifact."
+                },
+                "environment": {
+                    "description": "The environment into which this artifact is deployed. (Default - no envrionment.)",
+                    "type": "string"
+                },
+                "metadata": {
+                    "description": "Associated metadata. (Default - no metadata.)",
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "array",
+                        "items": {
+                            "$ref": "#/definitions/MetadataEntry"
+                        }
+                    }
+                },
+                "dependencies": {
+                    "description": "IDs of artifacts that must be deployed before this artifact. (Default - no dependencies.)",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "properties": {
+                    "description": "The set of properties for this artifact (depends on type) (Default - no properties.)",
+                    "type": "object",
+                    "additionalProperties": {}
+                }
+            },
+            "additionalProperties": false,
+            "required": [
+                "type"
+            ]
+        },
+        "ArtifactType": {
+            "description": "Type of cloud artifact.",
+            "enum": [
+                "aws:cloudformation:stack",
+                "cdk:tree",
+                "none"
+            ],
+            "type": "string"
+        },
+        "MetadataEntry": {
+            "description": "An metadata entry in the construct.",
+            "type": "object",
+            "properties": {
+                "type": {
+                    "description": "The type of the metadata entry.",
+                    "type": "string"
+                },
+                "data": {
+                    "description": "The data. (Default - no data.)",
+                    "anyOf": [
+                        {
+                            "description": "Any form of data. (for backwards compatibility to version < 1.31.0)"
```

Comment:
Nit: This is not for backwards compatibility. This is actually going to be the only option once we get rid of the abuse we've been doing on this attribute (i.e. all the rest are for backwards compatibility).

Reply:
makes sense